x-pack/plugins/license_management/index.js

```diff
@@ -21,9 +23,11 @@ export function licenseManagement(kibana)  {
       ]
     },
     init: (server) => {
-      registerLicenseRoute(server);
-      registerStartTrialRoutes(server);
-      registerStartBasicRoute(server);
+      registerLicenseChecker(server, PLUGIN.ID);
```

Comment:
It looks like these endpoints only need `xPackInfo`, as opposed to the entire `server`. Can we refactor this so that we're only passing the data required by each of these endpoints?

```js
    init: (server) => {
      registerLicenseChecker(server, PLUGIN.ID);
      const router = createRouter(server, PLUGIN.ID, '/api/license');
      const xPackInfo = server.plugins.xpack_main.info;
      registerLicenseRoute(router, xPackInfo);
      registerStartTrialRoutes(router, xPackInfo);
      registerStartBasicRoute(router, xPackInfo);
    }
```

Reply:
done